ci/src/main/scala/org/typelevel/sbt/CrossRootProject.scala

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.sbt
+
+import sbt._
+import org.scalajs.sbtplugin.ScalaJSPlugin
+import sbtghactions.GenerativePlugin.autoImport._
+import scala.scalanative.sbtplugin.ScalaNativePlugin
+
+/**
+ * Simultaneously creates a `root`, `rootJVM`, `rootJS`, and `rootNative` project, and
+ * automatically enables the `NoPublishPlugin`.
+ */
+final class CrossRootProject private (
+    val root: Project,
+    val rootJVM: Project,
+    val rootJS: Project,
+    val rootNative: Project
+) extends CompositeProject {
+
+  override def componentProjects: Seq[Project] = Seq(root, rootJVM, rootJS, rootNative)
+
+  def settings(ss: Def.SettingsDefinition*): CrossRootProject =
+    new CrossRootProject(
+      root.settings(ss: _*),
+      rootJVM.settings(ss: _*),
+      rootJS.settings(ss: _*),
+      rootNative.settings(ss: _*)
+    )
+
+  def configure(transforms: (Project => Project)*): CrossRootProject =
+    new CrossRootProject(
+      root.configure(transforms: _*),
+      rootJVM.configure(transforms: _*),
+      rootJS.configure(transforms: _*),
+      rootNative.configure(transforms: _*)
+    )
+
+  def enablePlugins(ns: Plugins*): CrossRootProject =
+    new CrossRootProject(
+      root.enablePlugins(ns: _*),
+      rootJVM.enablePlugins(ns: _*),
+      rootJS.enablePlugins(ns: _*),
+      rootNative.enablePlugins(ns: _*)
+    )
+
+  def disablePlugins(ps: AutoPlugin*): CrossRootProject =
+    new CrossRootProject(
+      root.disablePlugins(ps: _*),
+      rootJVM.disablePlugins(ps: _*),
+      rootJS.disablePlugins(ps: _*),
+      rootNative.disablePlugins(ps: _*)
+    )
+
+  def aggregate(projects: CompositeProject*): CrossRootProject =
+    aggregateImpl(projects.flatMap(_.componentProjects): _*)
+
+  private def aggregateImpl(projects: Project*): CrossRootProject = {
+
+    val jsProjects = projects.filter(p => Plugins.satisfied(p.plugins, Set(ScalaJSPlugin)))
```

Comment:
Hmm, these filters don't work like I ~expected~ wanted them to—`Plugins.satisfied` is too strict.